akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WithPriorKnowledgeSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.http.scaladsl.model.{ HttpResponse, StatusCodes }
+import akka.http.scaladsl.{ Http, HttpConnectionContext, UseHttp2 }
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, HttpProtocols }
+import akka.stream.ActorMaterializer
+import java.util.Base64
+import akka.stream.scaladsl.{ Sink, Source, Tcp }
+import akka.testkit.AkkaSpec
+import akka.util.ByteString
+
+import scala.concurrent.Future
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.{ SinkQueue, Keep }
+
+class WithPriorKnowledgeSpec extends AkkaSpec("""
```

Comment:
The HTTP/2 internals are somewhat well tested in `Http2ServerSpec` (without adding the transport layers). I think we are mostly missing any kind of end-to-end tests for HTTP/2. I guess we could keep the test like this for now but create a ticket to evolve it into more end-to-end like tests that could reuse all the high-level testing tools from `Http2ServerSpec`.

Reply:
@jrudolph Agreed.